tests/test_scalars.py

```diff
@@ -1302,13 +1296,17 @@ def test_timestamp_cmp() -> None:
     lte = check(assert_type(c_dt_datetime <= ts, bool), bool)
     assert gt != lte
 
-    check(assert_type(c_datetimeindex > ts, np_ndarray_bool), np.ndarray, np.bool_)
-    check(assert_type(c_datetimeindex <= ts, np_ndarray_bool), np.ndarray, np.bool_)
-    check(assert_type(c_unknown_index > ts, np_ndarray_bool), np.ndarray, np.bool_)
-    check(assert_type(c_unknown_index <= ts, np_ndarray_bool), np.ndarray, np.bool_)
+    check(assert_type(c_datetimeindex > ts, np_1darray[np.bool]), np_1darray[np.bool])
+    check(assert_type(c_datetimeindex <= ts, np_1darray[np.bool]), np_1darray[np.bool])
+    check(assert_type(c_unknown_index > ts, np_1darray[np.bool]), np_1darray[np.bool])
+    check(assert_type(c_unknown_index <= ts, np_1darray[np.bool]), np_1darray[np.bool])
 
     check(assert_type(c_np_ndarray_dt64 > ts, np_ndarray_bool), np.ndarray, np.bool_)
     check(assert_type(c_np_ndarray_dt64 <= ts, np_ndarray_bool), np.ndarray, np.bool_)
```

Comment:
why aren't these `np.1darray[np.bool]` ?

Reply:
@Dr-Irv I don't know what `np.1darray[np.bool]` is as it doesn't exist!
Do you mean `np_1darray[np.bool]` (1D array of bool) or `np_ndarray_bool` (array of bool of unknown dimension)?

If you mean the former then it wouldn't be correct. When you compare a scalar timestamp to arbitrary shape array, you get arbitrary shape array of bool, not 1D array of bool.

If you mean the latter, then OK I can change it (but we already have `np.bool` passed to `check` so it is already accounted for in a sense)

Reply:
I mean `np_1darray[np.boll]`.  I suggested adding tests where we compare the result of `to_numpy()` which are 1D to make sure we get 1D results back.  That will cover this issue.